### PR TITLE
Allow for custom component name when lowering to FuTIL.

### PR DESCRIPTION
**Pull request by cgyurgyik**
This passes in the kernel name of the configuration instead of a default string `"main"`.
So, when 
```
 ./fuse ./../futil/examples/dahlia/vectorized-add.fuse -n NewName -b futil --lower
```
is called, it produces: 
```
...
component NewName() -> () { ... }
``` 

However, it also means that the default would be "kernel" rather than "main," so what should we do here?

### CODE REVIEW

**Comment by sampsyo** [comment]
That's a good question! The hacky solution, which isn't so bad, would be to make the default different for C++ vs. FuTIL… i.e., the configuration would indicate whether we're in the default state, and the backend would use that to choose `kernel` (in `CppBackend`) or `main` (in `FutilBackend`).

I guess the alternative is to change Fud to invoke the Dahlia compiler with an explicit function name, but that seems like more hassle than it's worth.

**Comment by cgyurgyik** [comment]
> That's a good question! The hacky solution, which isn't so bad, would be to make the default different for C++ vs. FuTIL… i.e., the configuration would indicate whether we're in the default state, and the backend would use that to choose `kernel` (in `CppBackend`) or `main` (in `FutilBackend`).
> 
> I guess the alternative is to change Fud to invoke the Dahlia compiler with an explicit function name, but that seems like more hassle than it's worth.

I'm not familiar at all with Scala, so did something even worse...

**Comment by sampsyo** [comment]
Seems legit! Ready for an @sgpthomas review, perhaps?

File: src/main/scala/backends/futil/FutilBackend.scala [hunk 1]
```diff
@@ -651,7 +651,7 @@ private class FutilBackendHelper {
       "prog",
       List(
         Import("primitives/std.lib"),
-        Component("main", List(), List(), struct.sorted, control)
+        Component(if (c.kernelName == "kernel") "main" else c.kernelName, List(), List(), struct.sorted, control)
```
Comment (sgpthomas) [inline src/main/scala/backends/futil/FutilBackend.scala:654]: Hmm I dont like doing this. Does anything break in FuTIL if we just let `kernel` be the default name for components?

**Review by sgpthomas — CHANGES_REQUESTED** [review]

File: src/main/scala/backends/futil/FutilBackend.scala (hunk 1 above)
Comment (sampsyo) [reply src/main/scala/backends/futil/FutilBackend.scala:654]: @sgpthomas, as I noted in https://github.com/cucapra/dahlia/pull/324#issuecomment-720163384, I think we would have to change all the FuTIL tests that use Dahlia, and maybe fud itself, to use the right name?

Comment (sgpthomas) [reply src/main/scala/backends/futil/FutilBackend.scala:654]: when I get back to a computer I'll test this. But I don't think much needs to change. I think we would just have to change the Dahlia frontend tests and then the name that the testbench looks for

Comment (sampsyo) [reply src/main/scala/backends/futil/FutilBackend.scala:654]: Sounds legit.

Comment (sgpthomas) [reply src/main/scala/backends/futil/FutilBackend.scala:654]: yeah, we would have to change a lot of tests. I guess that we can do this for now.

**Review by sgpthomas — APPROVED** [review]

**Pull request merged**